flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/models/NotificationDetails.java

```diff
@@ -168,7 +169,10 @@ public class NotificationDetails implements Serializable {
   public Integer ledOffMs;
   public String ticker;
   public Integer visibility;
-  public Boolean allowWhileIdle;
+
+  @SerializedName(value = "scheduleMode", alternate = "allowWhileIdle")
```

Comment:
If i'm understanding this correctly, to handle the existing scheduled notifications that would've used `allowWhileIdle` that's a boolean you use this annotation to be ever to handle those values. When the plugin deserialises the old data, an exception will occur and this is where you can see if this data was created prior to this PR by checking if it's a boolean value and handle it accordingly. Does that sound about right? If so, that's pretty clever and learnt something new with GSON :)